src/Psl/Fun/after.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Psl\Fun;
+
+/**
+ * Returns a function that calls the next function with the result of the first function.
+ *
+ * Example:
+ *
+ *      $runExpression = Fun\after(
+ *          fn($i) => $i +  1,
+ *          fn(int $i) => $i * 5
+ *      );
+ *      => $runExpression(1) === 10;
+ *      => $runExpression(2) === 15;
+ *      => $runExpression(3) === 20;
+ *
+ *      $greet = Fun\after(
+ *          fn($value) => 'Hello' . $value,
+ *          fn($value) => $value . '!'
+ *      );
+ *      => $greet('World') === 'Hello World!';
+ *      => $greet('Jim') === 'Hello Jim!';
+ *
+ * @template I
+ * @template O
+ * @template R
+ *
+ * @psalm-param callable(I): O $first
+ * @psalm-param callable(O): R $next
+ *
+ * @psalm-return callable(I): R
+ *
+ * @psalm-pure
+ */
+function after(callable $first, callable $next): callable
+{
+    return fn ($input) => $next($first($input));
```

Comment:
```suggestion
    return static fn ($input) => $next($first($input));
```

Reply:
Since this is not in a class context (so no `$this`), does adding the "static" keyword actually cause any benefit?

Reply:
No, but more to do with coding style, and getting used to `static fn` over `fn`